server/datastore/mysql/hosts.go

```diff
@@ -712,9 +712,19 @@ func (ds *Datastore) EnrollHost(ctx context.Context, osqueryHostID, nodeKey stri
 // LoadHostByNodeKey loads the whole host identified by the node key.
 // If the node key is invalid it returns a NotFoundError.
 func (ds *Datastore) LoadHostByNodeKey(ctx context.Context, nodeKey string) (*fleet.Host, error) {
-	sqlStatement := `SELECT * FROM hosts WHERE node_key = ?`
+	// LoadHostByNodeKey is one of the most used queries (every time a host checks in),
+	// thus prepare such query once.
+	ds.prepLoadHostQueryOnce.Do(func() {
+		var err error
+		//nolint ds.loadHostStmt is closed in Datastore.Close
+		ds.loadHostStmt, err = sqlx.PreparexContext(ctx, ds.reader, `SELECT * FROM hosts WHERE node_key = ?`)
+		if err != nil {
+			panic(fmt.Sprintf("prepare LoadHostByNodeKey statement: %s", err))
```

Comment:
Not sure how I feel about a panic at this point. Can we prepare in the constructor if we want to panic?

Reply:
Also, without setting this up in an init style place... wouldn't there be a potential race with `ds.loadHostStmt`?

Reply:
Not easy, because at the time you call `sqlx.PreparexContenxt`, it needs to know about the `hosts` table.
(At the `Datastore.New` that is not the case, migrations haven't run yet.)

Reply:
Hm... I see... well, let's see if the results are worthwhile this weirdness.

Reply:
> Also, without setting this up in an init style place... wouldn't there be a potential race with ds.loadHostStmt?

According to the docs, we should be ok: https://pkg.go.dev/sync#Once.Do
`[...] Because no call to Do returns until the one call to f returns, if f causes Do to be called, it will deadlock.`

I also ran with -race just in case

Reply:
So I guess the risk is running Close while this is running... but doesn't sound like it matters.